charts/kube-prometheus-stack/values.yaml

```diff
@@ -876,10 +876,10 @@ kubeControllerManager:
     ## Enable scraping kube-controller-manager over https.
     ## Requires proper certs (not self-signed) and delegated authentication/authorization checks
     ##
-    https: false
+    https: true
 
     # Skip TLS certificate validation when scraping
-    insecureSkipVerify: null
+    insecureSkipVerify: true
```

Comment:
not sure if this should be true by default?
is there no option to add the root ca cert so the request will work without disabling verification complelty?

Reply:
Unfortunately, I didn't find a way to do that.

Root ca already in prometheus container and we can find it at /var/run/secrets/kubernetes.io/serviceaccount/ca.crt And the chart already use it for https connections: https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/exporters/kube-scheduler/servicemonitor.yaml#L28

If I set `insecureSkipVerify: false`, I get this error:
```
Get "https://10.234.3.101:10259/metrics": x509: certificate is valid for 127.0.0.1, not 10.234.3.101
```

10.23.3.101 -- a master node with Kube-scheduler.

If I set `tlsConfig.serverName` to 127.0.0.1 to fix this, I get another one:
```
Get "https://10.234.3.101:10259/metrics": x509: certificate signed by unknown authority
```

So, as I understand, kube-scheduler uses it's own certificate which doesn't signed with root cluster ca.

Reply:
May be that's the reason why insecure enabled in kube-prometheus chart too: https://github.com/prometheus-operator/kube-prometheus/blob/master/manifests/prometheus-serviceMonitorKubeScheduler.yaml#L15

Reply:
Oh, I guess they are just don't know too :)) https://github.com/prometheus-operator/kube-prometheus/issues/469

Reply:
Not sur if it works. 
Just remembered that i've used it like "curl --cacert ca.crt..." to speak to the kube api on port 6443 without cert error.